lib/flame/dispatcher/routes.rb

```diff
@@ -8,9 +8,14 @@ module Routes
 
 			## Find route and try execute it
 			def try_route
-				route = @app_class.router.find_route(request.path, request.http_method)
+				http_method = request.http_method
+				http_method = :GET if http_method == :HEAD
+				return nil unless @endpoint
+				allow = @endpoint.allow
+				halt(405, nil, 'Allow' => allow) if allow !~ /\b#{http_method}\b/
+				route = @endpoint[http_method]
```

Comment:
Maybe `halt 405` will be better after trying to find a route by specific HTTP-method? If route not found.

Reply:
`@endpoint` existence meaning that there is a nested routes or route here. so if we are not allowed for a HTTP-method, why should we to try find route with this method? 'allow' is returning HTTP-methods, that exist, so, if there is not allowed HTTP-method , why should we try find route by him?

Reply:
`Allow` is the server response header about what HTTP-methods are allowed, in case the requested HTTP-method is not found. `Routes#allow` returns a string composed of the available keys, and then you search for the item in this string — this is a strange "back and forth" action.

I see a logic like this: there is a request, there is the `Dispatcher#try_route` method, if the route not found with requested HTTP-method, but there are others with it — error (`halt`) with `Allow` header.

You first build a `halt`, and then check how much it's appropriate. It seems to me strange and not very obvious (perhaps not very simple).

Reply:
>  if the route not found with requested HTTP-method, but there are others

i trying to say that `available_endpoint` existence means that there is a routes or route. So we know, that there is a route or routes and  have to see whether the HTTP-method is allowed. If HTTP-method not allowed then `405`, else - execute route

Reply:
> i trying to say that available_endpoint existence means that there is a routes or route.

`Route` is not possible there. `Routes` or `nil`.

-----

I'm trying to say that code without `Regexp` may be simpler :)

```ruby
allow = available_endpoint.allow
halt(405, nil, 'Allow' => allow) if allow !~ /\b#{http_method}\b/
status 200
execute_route available_endpoint[http_method]
```

vs.

```ruby
route = available_endpoint[http_method]
halt(405, nil, 'Allow' => available_endpoint.allow) unless route
status 200
execute_route(route)
```

Reply:
where is `allow !~ /\b#{http_method}\b/` in your case? Or how you want to avoid it? Make `#allow` able accept arguments?
and..
```ruby
route = available_endpoint[http_method]
halt(405, nil, 'Allow' => available_endpoint.allow) unless route
status 200
execute_route(route)
```
not rigth.
As were said, `available_endpoint` may return `nil`
```
route = available_endpoint&.[](http_method)
halt(405, nil, 'Allow' => available_endpoint&.allow) unless route
status 200
execute_route(route)
```

Reply:
Uh. oK. understood. Changed.